### PR TITLE
AppSec extension & helper unit tests in CI

### PR DESCRIPTION
**Pull request by Anilm3**
### Description

This PR introduces:
- Extension unit tests for all versions of PHP (including 8.3), debug, debug-zts and debug-zts-asan variants,  x86_64 and aarch64.
- Extension and helper test and coverage for x86_64 and aarch64
- Extension and helper lint (only x86_64)
- Helper asan tests for x86_64 and aarch64

Some issues have been identified:
- Unfortunately PHP 8.0 with debug-zts-asan and aarch64 has some static TLS issues  which I can only resolve by reordering libraries, so I've skipped it for now.
- The XML coverage is failing to upload to codecov, but we're also uploading an HTML coverage report to the artifacts tab.

Related Jiras: [APPSEC-11709], [APPSEC-11710], [APPSEC-10149]
### Readiness checklist
- [ ] (only for Members) Changelog has been added to the release document.
- [ ] Tests added for this feature/bug.

### Reviewer checklist
- [ ] Appropriate labels assigned.
- [ ] Milestone is set.
- [ ] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

[APPSEC-11709]: https://datadoghq.atlassian.net/browse/APPSEC-11709?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APPSEC-11710]: https://datadoghq.atlassian.net/browse/APPSEC-11710?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APPSEC-10149]: https://datadoghq.atlassian.net/browse/APPSEC-10149?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: appsec/src/extension/helper_process.c [hunk 1]
```diff
@@ -522,8 +524,8 @@ static char **nullable _split_params(
     }
 
     // we never write more than the original size of the params
-    char *params_buffer = emalloc(strlen(orig_params_str) + 1);
-    char *wp = params_buffer; // write pointer
+    char *params_buffer = emalloc(strlen(orig_params_str) + 1); // NOLINT
+    char *wp = params_buffer;                                   // write pointer
```
Comment (estringana) [inline appsec/src/extension/helper_process.c:528]: Wrong linting

**Review by estringana — APPROVED** [review]

File: appsec/src/extension/helper_process.c (hunk 1 above)
Comment (Anilm3) [reply appsec/src/extension/helper_process.c:528]: What do you mean?

Comment (estringana) [reply appsec/src/extension/helper_process.c:528]: Line 528 is aligning the comment with the one on the previous line. However the one on the following is not. So I'd say, it should not be aligned

Comment (Anilm3) [reply appsec/src/extension/helper_process.c:528]: clang-format quirk I suppose

**Review by bwoebi — APPROVED** [review]

**Pull request merged**